shotover-proxy/src/transforms/cassandra/sink_cluster.rs

```diff
@@ -271,69 +290,70 @@ async fn topology_task_process(
-        if let CassandraOperation::Result(CassandraResult::Rows {
-            value: MessageValue::Rows(rows),
-            metadata,
-        }) = &mut frame.operation
-        {
-            for row in rows.iter() {
-                let mut address = None;
-                let mut rack = None;
-                let mut data_center = None;
-                let mut tokens = vec![];
-                for (i, col) in metadata.col_specs.iter().enumerate() {
-                    let ident = Identifier::parse(&col.name);
-                    if ident == peer_ident {
-                        if let Some(MessageValue::Inet(value)) = row.get(i) {
-                            address = Some(*value);
-                        }
-                    } else if ident == rack_ident {
-                        if let Some(MessageValue::Varchar(value)) = row.get(i) {
-                            rack = Some(value.clone());
-                        }
-                    } else if ident == data_center_ident {
-                        if let Some(MessageValue::Varchar(value)) = row.get(i) {
-                            data_center = Some(value.clone());
-                        }
-                    } else if ident == tokens_ident {
-                        if let Some(MessageValue::List(list)) = row.get(i) {
-                            tokens = list
-                                .iter()
-                                .filter_map(|x| match x {
-                                    MessageValue::Varchar(a) => Some(a.clone()),
-                                    _ => None,
-                                })
-                                .collect();
-                        }
+        match &mut frame.operation {
+            CassandraOperation::Result(CassandraResult::Rows {
+                value: MessageValue::Rows(rows),
+                ..
+            }) => rows
+                .iter_mut()
+                .filter(|row| {
+                    if let Some(MessageValue::Varchar(data_center)) = row.get(2) {
+                        data_center == config_data_center
+                    } else {
+                        false
                     }
-                }
-                if let (Some(address), Some(rack), Some(data_center)) = (address, rack, data_center)
-                {
-                    if data_center == config_data_center {
-                        new_nodes.push(CassandraNode {
-                            address,
-                            _rack: rack,
-                            _tokens: tokens,
-                            outbound: None,
-                        });
+                })
+                .map(|row| {
+                    if row.len() != 4 {
+                        return Err(anyhow!("expected 4 columns but was {}", row.len()));
                     }
-                }
-            }
+
+                    let tokens = if let Some(MessageValue::List(list)) = row.pop() {
```

Comment:
This is actually a bug in cassandra-protocol, the tokens should be a set. Happy to move ahead with as is, but once https://github.com/krojew/cdrs-tokio/pull/103 is merged and we update cassandra-protocol this will need to be fixed.

Reply:
oooh good catch, yeah lets merge and fix it once the cdrs-tokio fix lands